env/env.cc

```diff
@@ -618,13 +638,63 @@ Status Env::LoadEnv(const std::string& value, Env** result,
     *result = env;
   }
 #else
+  (void)config_options;
   (void)result;
   (void)guard;
   s = Status::NotSupported("Cannot load environment in LITE mode", value);
 #endif
   return s;
 }
 
+Status Env::CreateFromSystem(const ConfigOptions& config_options, Env** result,
+                             std::shared_ptr<Env>* guard) {
+  *result = config_options.env;
+  const char* env_uri = getenv("TEST_ENV_URI");
+  const char* fs_uri = getenv("TEST_FS_URI");
+  if (env_uri && fs_uri) {  // Both specified.  Cannot choose.  Return Invalid
+    return Status::InvalidArgument("cannot specify both fs_uri and env_uri");
+  } else if (env_uri) {  // Only have an ENV URI.  Create an Env from it
+    return CreateFromString(config_options, env_uri, result, guard);
+  } else if (fs_uri) {  // Only have an FS URI.  Create an FS and wrap it
+    std::shared_ptr<FileSystem> fs;
+    Status s = FileSystem::CreateFromString(config_options, fs_uri, &fs);
+    if (s.ok()) {
+      guard->reset(new CompositeEnvWrapper(*result, fs));
+      *result = guard->get();
+    }
+    return s;
+  } else {
+    // Neither specified.  Use the default
+    guard->reset();
+    return Status::OK();
+  }
+}
+
+Status Env::CreateFromFlags(const ConfigOptions& config_options,
+                            const std::string& env_uri,
+                            const std::string& fs_uri, Env** result,
+                            std::shared_ptr<Env>* guard) {
+  *result = config_options.env;
+  if (env_uri.empty() && fs_uri.empty()) {
```

Comment:
Can we reuse the code of CreateFromSystem. The logic is almost the same.

Reply:
Done.

Reply:
I mean, remove the CreateFromSystem method. Using CreateFromArgv or others to achieve the same function. I cannot see the essential of introducing CreateFromSystem.

Reply:
Moved it to test::CreateEnvFromSystem and out of the public API